### PR TITLE
feat: install vault inside k3s cluster 

### PR DESCRIPTION
**Pull request by babadofar**
Added infra folder with helmfile install for Vault inside the k3s cluster

### CODE REVIEW

**Comment by keskad** [comment]
How does it relate to external vault option?

File: infra/helmfile.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+filepath: ""
```
Comment (ankitm123) [inline infra/helmfile.yaml]: let's rename infra to jx-vault, I think we follow the convention where the folder name is the namespace

File: infra/values.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+storage:
```
Comment (ankitm123) [inline infra/values.yaml]: Why this file? let vault generate it's own creds, and then we link the docs where we describe how to get the root credentials: https://jenkins-x.io/v3/admin/setup/secrets/vault/#using-the-vault-cli-directly

**Review by ankitm123 — COMMENTED** [review]
few comments.

File: infra/helmfile.yaml (hunk 1 above)
Comment (babadofar) [reply infra/helmfile.yaml]: Good idea

File: infra/values.yaml (hunk 2 above)
Comment (babadofar) [reply infra/values.yaml]: Ok thanks

**Comment by babadofar** [comment]
External Vault is configured outside the cluster, so,they are related in the sense Vault is used for as secure store, but the install method will be completely different

File: infra/values.yaml (hunk 2 above)
Comment (babadofar) [reply infra/values.yaml]: Well actually the idea with adding a values file was that I wanted it to use in-memory storage and development mode, to make it more light weight. But I can definitely remove the devRootToken

Comment (babadofar) [reply infra/values.yaml]: Vault seems to require 10G storage as default. https://github.com/hashicorp/vault-helm/blob/main/values.yaml#L590

**Comment by ankitm123** [comment]
> How does it relate to external vault option?

This is just vault getting installed inside the k3s cluster. For external vault, we have to update the documents for end user.

File: infra/values.yaml (hunk 2 above)
Comment (ankitm123) [reply infra/values.yaml]: that makes sense, may be set the default storage to 1? I am not sure what is a good default?

**Comment by ankitm123** [comment]
may be change the PR title and commit message to `feat: install vault inside k3s cluster` - local might be confusing, as external vault is also local to the workstation, but external to the kubernetes cluster 🤔 , wdyt?

File: infra/values.yaml (hunk 2 above)
Comment (babadofar) [reply infra/values.yaml]: Ok, lets try that. I guess it makes it easier for people if vault remembers the secrets between runs.

File: .jx/secret/mapping/secret-mappings.yaml [hunk 3]
```diff
@@ -56,44 +59,43 @@ spec:
...
     mappings:
-    - name: azure.json
-      key: secret/data/externaldns
+    - key: secret/data/externaldns
+      name: azure.json
       property: azure.json
-  defaults:
-    backendType: vault
\ No newline at end of file
+    name: external-dns
```
Comment (ankitm123) [inline .jx/secret/mapping/secret-mappings.yaml:101]: Why external dns?

File: config-root/cluster/namespaces/jx-staging.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
```
Comment (ankitm123) [inline config-root/cluster/namespaces/jx-staging.yaml:1]: Dont add config root to the PR.

File: docs/releases.yaml [hunk 5]
```diff
@@ -1,6 +1,122 @@
 - namespace: jx
   path: helmfiles/jx/helmfile.yaml
+  releases:
```
Comment (ankitm123) [inline docs/releases.yaml:3]: dont need these changes as well - look at jx3-eks-vault template repo.

File: docs/README.md [hunk 6]
```diff
@@ -11,6 +11,63 @@
     </tr>
   </thead>
   <tbody>
+    <tr>
```
Comment (ankitm123) [inline docs/README.md:14]: dont need these changes I think - check the eks-vault template repo.

**Pull request closed without merging**